### DISCUSSION TITLE
How to report metadata back to host from the guest?

### DISCUSSION THREAD

**Discussion by jlkiri** [firecracker-microvm/firecracker#3834] [Q&A] [answered]
I have a simple use case where I setup a DHCP server on the bridge to which TAP devices are attached so all of my VMs get an IP address. However the code that is responsible for launching VMs has no way to know the IP that the VM got. I looked at MMDS but it looks like you can't update it from inside the guest. What would be the best way to report VM's IP address back to the host?

**Comment by kalyazin** [comment, marked as answer]
Hi @jlkiri . Did you consider using [vsock](https://github.com/firecracker-microvm/firecracker/blob/main/docs/vsock.md) for that?

**Reply by jlkiri** [reply]
Yes I think I can do that.
